### DISCUSSION TITLE
Download Table as CSV for locally saved reports

### DISCUSSION THREAD

**Discussion by pitgreisen** [datapane/datapane#197] [Ideas]
Hi datapane team,

One feature that I found in the doc but that I am not able to reproduce is the possibility to download a table as csv/xlsx, especially for locally saved reports. I understand this is currently not supported?

Is it possible to add this feature in a future version?

Thank you!

**Comment by Buroni** [comment]
Hey @pitgreisen, you're right that all the datatable extension functions are disabled locally.

The CSV downloading is all client-side, so we could look into enabling this locally. However the excel export is handled server-side, so support for this wouldn't be possible in the near future.

**Reply by pitgreisen** [reply]
Hi @Buroni,  

Thank you.  I wrote CSV *and* excel, but I am really mostly interested in the local CSV option. If that can be enabled in the near future, that would be super helpful for my use case. Even better with the current state/original data download option.

**Reply by Buroni** [reply]
We're currently re-writing the report component -- I've added the "Current state" export options to the local report so that will be available when the rewritten component is released.  

A local "Original data" option would have the downside that we'd need to store both the current state and the original data object in memory, rather than keeping the current state in memory and pulling the original from the server where needed as we do with remote reports. So I've kept this disabled locally for now.
